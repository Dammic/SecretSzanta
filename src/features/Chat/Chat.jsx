import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import MessagesBox from './MessagesBox/MessagesBox'
import ChatForm from './ChatForm/ChatForm'

import styles from './Chat.css'

const Chat = ({
    className,
}) => (
    <div className={classNames(styles.chat, className)}>
        <MessagesBox />
        <ChatForm />
    </div>
)

Chat.displayName = 'Chat'
Chat.propTypes = {
    className: PropTypes.string,
}
export default Chat


