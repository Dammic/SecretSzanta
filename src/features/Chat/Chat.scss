.chat {
    width: 400px;
    height: 600px;
    display: flex;
    flex-direction: column;

    .messages-box {
        width: 100%;
        height: 100%;
    }

    .chat-form {
        width: 100%;
        height: 100px;
        max-height: 100px;
    }

}