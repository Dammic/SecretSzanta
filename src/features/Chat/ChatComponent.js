'use strict'
import React from 'react'
import MessagesBox from './MessagesBox/MessagesBox'
import ChatForm from './ChatForm/ChatForm'

const ChatComponent = ({
    socket
}) => {
    return (
        <div className="chat">
            <MessagesBox socket={socket} />
            <ChatForm socket={socket} />
        </div>
    )
}

export default ChatComponent


