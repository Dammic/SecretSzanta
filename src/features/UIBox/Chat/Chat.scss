.chat {
    width: 350px;
    height: 20%;
    display: flex;
    flex-direction: column;
    right: 2px;
    position: absolute;
    bottom: 0;
    z-index: 30;
}
