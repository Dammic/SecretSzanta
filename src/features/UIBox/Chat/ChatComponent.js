'use strict'
import React from 'react'
import MessagesBox from './MessagesBox/MessagesBox'
import ChatForm from './ChatForm/ChatForm'

const ChatComponent = ({
    userName
}) => {
    return (
        <div className="chat">
            <MessagesBox />
            <ChatForm userName={userName}/>
        </div>
    )
}

export default ChatComponent

