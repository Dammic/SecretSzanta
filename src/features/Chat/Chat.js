'use strict'
import React from 'react'
import ChatComponent from './ChatComponent'

export default class Chat extends React.PureComponent {
    render () {
        const {socket} = this.props
        const {userName} = this.props
        return (
            <ChatComponent socket={socket} userName={userName}/>
        )
    }
}
