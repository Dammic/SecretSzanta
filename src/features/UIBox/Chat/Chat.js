'use strict'
import React from 'react'
import ChatComponent from './ChatComponent'

export default class Chat extends React.PureComponent {
    render () {
        const {userName} = this.props
        return (
            <ChatComponent userName={userName}/>
        )
    }
}
