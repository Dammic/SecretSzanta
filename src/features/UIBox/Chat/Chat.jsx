import React from 'react'
import PropTypes from 'prop-types'
import ChatComponent from './ChatComponent'

export default class Chat extends React.PureComponent {
    static propTypes = {
        // parent
        userName: PropTypes.string,
        className: PropTypes.string,
    }

    render() {
        const { userName, className } = this.props
        return (
            <ChatComponent userName={userName} className={className} />
        )
    }
}
