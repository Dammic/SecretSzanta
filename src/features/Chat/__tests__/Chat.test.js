import React from 'react'
import { shallow } from 'enzyme'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import Chat from '../Chat'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        userName: 'mockUserName',
        className: 'mockClassName',
        ...propsOverrides,
    }
    const component = renderMethod(<Chat { ...props } />)
    return { props, component }
}

describe('<Chat />', () => {
    it('matches snapshot', () => {
        const { props } = setupProps()
        expectShallowMatchingSnapshot(<Chat {...props} />)
    })
})
