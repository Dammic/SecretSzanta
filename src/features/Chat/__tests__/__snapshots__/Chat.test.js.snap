// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> matches snapshot 1`] = `
<div
  className="chat mockClassName"
>
  <Connect(MessagesBox) />
  <Connect(ChatForm) />
</div>
`;
